File: app/src/main/java/com/qiaoqiao/core/camera/awareness/map/PlaceWrapper.java
package com.qiaoqiao.core.camera.awareness.map;


import android.graphics.Bitmap;

import com.google.android.gms.location.places.Place;
import com.google.android.gms.maps.model.LatLng;
import com.google.maps.android.clustering.ClusterItem;

public final class PlaceWrapper implements ClusterItem{
	private final Place mPlace;
	private final Bitmap mBitmap;

	public PlaceWrapper(Place place, Bitmap bitmap) {
		mPlace = place;
		mBitmap = bitmap;
	}


	public Place getPlace() {
		return mPlace;
	}

	public Bitmap getBitmap() {
		return mBitmap;
	}

	@Override
	public LatLng getPosition() {
		return mPlace.getLatLng();
	}

	@Override
	public String getTitle() {
		return mPlace.getName().toString();
	}

	@Override
	public String getSnippet() {
		return null;
	}
}

File: app/src/main/java/com/qiaoqiao/core/camera/awareness/map/PlaceWrapper.kt
package com.qiaoqiao.core.camera.awareness.map

import android.graphics.Bitmap
import com.google.android.gms.location.places.Place
import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.clustering.ClusterItem
import com.qiaoqiao.customtabs.CustomTabUtils
import java.io.Serializable

class PlaceWrapper(val place: Place, val bitmap: Bitmap?) : ClusterItem, Serializable {
    init {
        CustomTabUtils.HELPER.mayLaunchUrl(place.websiteUri, null, null)
    }

    override fun getPosition(): LatLng {
        return place.latLng
    }

    override fun getTitle(): String {
        return place.name.toString()
    }

    override fun getSnippet(): String? {
        return null
    }
}